### PR TITLE
feat(gcb): Add Google Cloud Build stage

### PR DESCRIPTION
**Pull request by ezimanyi**
Add a stage that triggers a build using Google Cloud Build. At this point, the stage only accepts a build definition directly in the stage, and does not wait for the build to complete.

### CODE REVIEW

File: orca-igor/orca-igor.gradle [hunk 1]
```diff
@@ -22,6 +22,7 @@ dependencies {
   compile project(":orca-front50")
   compile spinnaker.dependency('kork')
   compile spinnaker.dependency('bootAutoConfigure')
+  compile 'com.google.apis:google-api-services-cloudbuild:v1-rev836-1.25.0'
```
Comment (jtk54) [inline orca-igor/orca-igor.gradle]: Removed when we bump spinnaker-dependencies?

File: orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/model/GoogleCloudBuildStageDefinition.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.api.services.cloudbuild.v1.model.Build;
+import lombok.Getter;
+
+@Getter
+public class GoogleCloudBuildStageDefinition {
+  private final String account;
+  private final Build buildDefinition;
+
+  // There does not seem to be a way to auto-generate a constructor using our current version of Lombok (1.16.20) that
```
Comment (jtk54) [inline orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/model/GoogleCloudBuildStageDefinition.java]: Jackson needs getters and setters for the class fields, so `@Data` on the class usually works here.

File: orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/pipeline/GoogleCloudBuildStage.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class GoogleCloudBuildStage implements StageDefinitionBuilder {
+  @Override
+  public void taskGraph(@Nonnull Stage stage, @Nonnull TaskNode.Builder builder) {
+    GoogleCloudBuildStageDefinition stageDefinition = stage.mapTo(GoogleCloudBuildStageDefinition.class);
+    builder
+      .withTask("startGoogleCloudBuildTask", StartGoogleCloudBuildTask.class);
```
Comment (jtk54) [inline orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/pipeline/GoogleCloudBuildStage.java]: Totally async right now -- will this eventually be enriched to parse build results or no?

File: orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/tasks/StartGoogleCloudBuildTask.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+@Component
+@RequiredArgsConstructor
+public class StartGoogleCloudBuildTask implements Task {
+  private final IgorService igorService;
+
+  @Override
+  @Nonnull public TaskResult execute(@Nonnull Stage stage) {
+    GoogleCloudBuildStageDefinition stageDefinition = stage.mapTo(GoogleCloudBuildStageDefinition.class);
+    Operation result = igorService.createGoogleCloudBuild(stageDefinition.getAccount(), stageDefinition.getBuildDefinition());
+    return new TaskResult(ExecutionStatus.SUCCEEDED);
```
Comment (jtk54) [inline orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/tasks/StartGoogleCloudBuildTask.java]: Should this be dependent on the build result?

Comment (jtk54) [reply orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/tasks/StartGoogleCloudBuildTask.java]: Or is that just the async tracking task in GCP?

File: orca-igor/orca-igor.gradle (hunk 1 above)
Comment (ezimanyi) [reply orca-igor/orca-igor.gradle]: Yup, exactly.  I'll remove it in this same PR after I rebase on top of the bump.

File: orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/model/GoogleCloudBuildStageDefinition.java (hunk 2 above)
Comment (ezimanyi) [reply orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/model/GoogleCloudBuildStageDefinition.java]: Yeah, I really wanted an immutable class as the stage definition shouldn't be mutated (and won't actually be persisted if you do mutate it)...and this was the best way I could find to have Jackson deserialize into an immutable class.

File: orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/pipeline/GoogleCloudBuildStage.java (hunk 3 above)
Comment (ezimanyi) [reply orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/pipeline/GoogleCloudBuildStage.java]: Yup, absolutely---adding support for getting build status to igor, then having a task in orca to wait for that is next on my list.

File: orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/tasks/StartGoogleCloudBuildTask.java (hunk 4 above)
Comment (ezimanyi) [reply orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/tasks/StartGoogleCloudBuildTask.java]: This task will succeed if the build is correctly triggered, then there will be a second "wait for results" task that will wait for the build to complete and succeed/fail based on the actual build result.

**Pull request merged**